go/src/github.com/mesosphere/rendler/results.go

```diff
@@ -5,48 +5,52 @@ import (
 	"fmt"
 )
 
-// The type of crawl results
+// CrawlResult is the result of a crawl operation
 type CrawlResult struct {
 	TaskID string
 	URL    string
 	Links  []string
 }
 
-func (target CrawlResult) FromJson(data []byte) CrawlResult {
-	err := json.Unmarshal(data, &target)
+// FromJSON converts a JSON blob to a CrawlResult
+func (c *CrawlResult) FromJSON(data []byte) error {
```

Comment:
I think these methods are completely redundant. We can just use the `json.Marshal` and `json.Unmarshal` directly.